web-search-engines/langchain4j-web-search-engine-searchapi/src/test/java/dev/langchain4j/web/search/searchapi/SearchApiWebSearchEngineIT.java

```diff
@@ -0,0 +1,122 @@
+package dev.langchain4j.web.search.searchapi;
+
+import static dev.langchain4j.web.search.searchapi.SearchApiWebSearchEngine.DEFAULT_ENGINE;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.List;
+import java.util.Map;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.condition.EnabledIfEnvironmentVariable;
+
+import dev.langchain4j.web.search.WebSearchEngine;
+import dev.langchain4j.web.search.WebSearchEngineIT;
+import dev.langchain4j.web.search.WebSearchInformationResult;
+import dev.langchain4j.web.search.WebSearchOrganicResult;
+import dev.langchain4j.web.search.WebSearchResults;
+
+@EnabledIfEnvironmentVariable(named = "SEARCHAPI_API_KEY", matches = ".+")
+class SearchApiWebSearchEngineIT extends WebSearchEngineIT {
+
+	public static final String SEARCHAPI_API_KEY = "SEARCHAPI_API_KEY";
+	
+	private static final boolean logRequests = false;
+	
+    protected WebSearchEngine webSearchEngine = SearchApiWebSearchEngine.withApiKey(System.getenv(SEARCHAPI_API_KEY)).logRequests(logRequests).build();
+
+    @Test
+    void should_search_with_defaults() {
```

Comment:
is there a need for this test? `WebSearchEngineIT.should_search` seems to test the same/similar

Reply:
You are right: `should_search_with_defaults()` duplicates the tests in `WebSearchEngineIT#should_search()`.

I'll remove it.

Reply:
Removed in [93c5feb](https://github.com/langchain4j/langchain4j/pull/1215/commits/93c5feb4ffcc41b8a5ad55c90d3366625659d4ac).